referencesrv/resolver/hypotheses.py

```diff
@@ -106,16 +112,23 @@ def make_digested_record(self):
             # remove too much information
             self.digested_record["title"] = self.TITLE_MAIN.split(self.digested_record["title"])[0]
 
-        if "pub" in self.digested_record:
+        pub = self.digested_record.get("pub", None) or self.digested_record.get("title", None)
+        if pub:
             try:
-                bibstem = get_best_bibstem_for(self.digested_record["pub"])
-                # if bibstem is one of the multi-section journal,
-                # sometimes user do not include the section char
-                # so included a wildcard in bibstem
-                if any([bib==bibstem for bib in self.BIBSTEM_WITH_SECTIONS]):
-                    self.digested_record["bibstem"] = '%s*'%bibstem
+                if len(pub) <= 2:
+                    if self.digested_record.get("pub", None):
+                        self.digested_record.remove("pub")
+                    elif self.digested_record.get("title", None):
+                        self.digested_record.remove("title")
                 else:
-                    self.digested_record["bibstem"] = bibstem
+                    bibstem = get_best_bibstem_for(pub)
+                    # if bibstem is one of the multi-section journal,
+                    # sometimes user do not include the section char
+                    # so included a wildcard in bibstem
+                    if any([bib==bibstem for bib in self.BIBSTEM_WITH_SECTIONS]):
+                        self.digested_record["bibstem"] = '%s*'%bibstem
+                    else:
+                        self.digested_record["bibstem"] = bibstem
             except:
                 self.digested_record["bibstem"] = ''
```

Comment:
ok, I know this has not been changed, but I'd have a suggestion. 

inside `get_best_bibstem` throw exception `BibstemException`

which is defined as

`class BibstemException(Exception): pass`

and then in the block here, do:

```try:
     .....
   except BibstemException as e:
      self.digested_record["bibstem"] = ''
```

that way, if a bug appears anywhere else; it will get to be known relatively soon; which will make any future coding life little bit more pleasant

Reply:
so you are suggesting to replace the exception from `KeyError` to `BibstemException`
https://github.com/golnazads/reference_service/blob/master/referencesrv/resolver/journalfield.py#L46

Reply:
sorry, what i was point at is the line 132; with the broad `except` catch - that's the one to become `except KeyError` (or BibstemException)